### ISSUE TITLE
[external]: using custom icons makes ownCloud not available after upgrading to 10.0.3

### ISSUE THREAD

**Issue by pako81** [owncloud/apps#2222]
### Affected apps
External sites

### Expected behaviour
When using the external app with custom icons (in /apps/external/img/) upgrading from 9.1.x to 10.0.3 makes ownCloud to be not accessible anymore

### Actual behaviour
ownCloud should be accessible after the upgrade

### Steps to reproduce
1. activate external app and put a custom icon on /apps/external/img/
2. be sure everything works correctly
3. upgrade from 9.1.x to 10.0.3 -> ownCloud not accessible anymore - Internal Server Error

### Server configuration
Operating system:
Ubuntu 16.04

Web server: 
Apache 2.4

Database:
MariaDB 10.0.31

PHP version:
7.0.22

ownCloud version:

10.0.3

#### ownCloud log (data/owncloud.log)
```
{"reqId":"GtUe6fLy3HLO0GTXzZ7a","level":3,"time":"2017-09-26T14:31:30+00:00","remoteAddr":"127.0.0.1","user":"pastripodi","app":"index","method":"GET","url":"\/owncloud1003\/index.php\/apps\/files\/","message":"Exception: {\"Exception\":\"RuntimeException\",\"Message\":\"image not found: image:icons8-Icons8-48.png webroot:\\\/owncloud1003 serverroot:\\\/var\\\/www\\\/html\\\/owncloud1003\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud1003\\\/apps\\\/external\\\/appinfo\\\/app.php(38): OC\\\\URLGenerator->imagePath('external', 'icons8-Icons8-4...')\\n#1 \\\/var\\\/www\\\/html\\\/owncloud1003\\\/lib\\\/private\\\/NavigationManager.php(95): OC_App::{closure}()\\n#2 \\\/var\\\/www\\\/html\\\/owncloud1003\\\/lib\\\/private\\\/legacy\\\/app.php(711): OC\\\\NavigationManager->getAll()\\n#3 \\\/var\\\/www\\\/html\\\/owncloud1003\\\/lib\\\/private\\\/TemplateLayout.php(76): OC_App::getNavigation()\\n#4 \\\/var\\\/www\\\/html\\\/owncloud1003\\\/lib\\\/private\\\/legacy\\\/template.php(232): OC\\\\TemplateLayout->__construct('user', 'files')\\n#5 \\\/var\\\/www\\\/html\\\/owncloud1003\\\/lib\\\/public\\\/AppFramework\\\/Http\\\/TemplateResponse.php(156): OC_Template->fetchPage()\\n#6 \\\/var\\\/www\\\/html\\\/owncloud1003\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(112): OCP\\\\AppFramework\\\\Http\\\\TemplateResponse->render()\\n#7 \\\/var\\\/www\\\/html\\\/owncloud1003\\\/lib\\\/private\\\/AppFramework\\\/App.php(98): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OCA\\\\Files\\\\Controller\\\\ViewController), 'index')\\n#8 \\\/var\\\/www\\\/html\\\/owncloud1003\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(46): OC\\\\AppFramework\\\\App::main('ViewController', 'index', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#9 [internal function]: OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n#10 \\\/var\\\/www\\\/html\\\/owncloud1003\\\/lib\\\/private\\\/Route\\\/Router.php(307): call_user_func(Object(OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler), Array)\\n#11 \\\/var\\\/www\\\/html\\\/owncloud1003\\\/lib\\\/base.php(928): OC\\\\Route\\\\Router->match('\\\/apps\\\/files\\\/')\\n#12 \\\/var\\\/www\\\/html\\\/owncloud1003\\\/index.php(56): OC::handleRequest()\\n#13 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud1003\\\/lib\\\/private\\\/URLGenerator.php\",\"Line\":164}"}
```

@DeepDiver1975 as discussed

**Comment by PVince81** [comment]
@phisch some resource locator madness involved ?

**Comment by cdamken** [comment]
@PVince81 Can you assign an engineer to this case?

**Comment by PVince81** [comment]
See https://github.com/owncloud/core/issues/20941

It is possible that the icon path isn't the same or not resolved properly.

The ticket I linked above was mostly for preventing to crash the whole server in case an image is not found but not fixing the path resolution.

Considering that we fixed theme path resolution in 10.0.8 I'd suggest trying an upgrade first and verifying that the icon path in question is still correct.
